File: program_data/ProgramData/90/2370.cpp
//********************************
//*????????    **
//*?????? 1300012887 **
//*???2013.11.27 **
//********************************
int count(int M, int N)  //??????????
{
	if (M == 1 || N == 1 || N == 0)  //??????????1??????0
		return 1;  //??1?
	if (M < N)  //??????????
		return count(M, M);  //??????????????????
	return (count(M, N-1) + count(M-N, N));  //????????????????????
}
int main()  //?????
{
	int t, i, a, n, m;
	cin >> t;  //????????
	cin.get();
	for (i = 0; i < t; i++)
	{
		cin >> m >> n;  //????????
		a = count (m, n);
		cout << a << endl;  //??????
	}
	return 0;
}

File: program_data/JavaProgramData/62/2370.java
package <missing>;

public class GlobalMembers
{
	public static int Main()
	{
		int i = 0;
		int j = 0;
		int count = 0;
		final String a = "";
		char c = System.in.read();
		while (c != '\n')
		{
			if (c == ' ')
			{
				if (count != 0)
				{
					c = System.in.read();
					continue;
				}
				else if (count == 0)
				{
					i++;
					j = 0;
					count++;
				}
			}
			else
			{
				a.charAt(i)[j++] = c;
				count = 0;
			}
			c = System.in.read();
		} //??????????????????????????????????????????
		count = i;
		for (i = 0;i < count;i++)
		{
			for (j = 0;a.charAt(i)[j] != '\0';j++)
			{
				System.out.print(a.charAt(i)[j]);
			}
			System.out.print(" ");
		}
		for (i = 0;a.charAt(count)[i] != '\0';i++)
		{
			System.out.print(a.charAt(count)[i]);
		}
		return 0;
	}
}
